src/socket_listener_win32.c

```diff
@@ -138,31 +137,28 @@ int socketlistener_stop(SOCKET_LISTENER_HANDLE socket_listener)
 
 void socketlistener_dowork(SOCKET_LISTENER_HANDLE socket_listener)
 {
-	if (socket_listener != NULL)
-	{
-		SOCKET_LISTENER_INSTANCE* socket_listener_instance = (SOCKET_LISTENER_INSTANCE*)socket_listener;
-		SOCKET accepted_socket = accept(socket_listener_instance->socket, NULL, NULL);
+    if (socket_listener == NULL)
+    {
+        LogError("NULL socket_listener");
+    }
+    else
+    {
+		SOCKET accepted_socket = accept(socket_listener->socket, NULL, NULL);
 		if (accepted_socket != INVALID_SOCKET)
 		{
-			if (socket_listener_instance->on_socket_accepted != NULL)
+			SOCKETIO_CONFIG socketio_config;
+            socketio_config.hostname = NULL;
+            socketio_config.port = socket_listener->port;
+            socketio_config.accepted_socket = &accepted_socket;
+            XIO_HANDLE io = xio_create(socketio_get_interface_description(), &socketio_config);
+			if (io == NULL)
 			{
-				SOCKETIO_CONFIG socketio_config;
-                socketio_config.hostname = NULL;
-                socketio_config.port = socket_listener_instance->port;
-                socketio_config.accepted_socket = &accepted_socket;
-                XIO_HANDLE io = xio_create(socketio_get_interface_description(), &socketio_config);
-				if (io == NULL)
-				{
-					(void)closesocket(accepted_socket);
-				}
-				else
-				{
-					socket_listener_instance->on_socket_accepted(socket_listener_instance->callback_context, io);
-				}
+                LogError("Could not create IO");
+                (void)closesocket(accepted_socket);
 			}
 			else
 			{
-				(void)closesocket(accepted_socket);
+				socket_listener->on_socket_accepted(socket_listener->callback_context, io);
```

Comment:
The event is on the fact that a socket was accepted. When a socket is accepted by the listener a new socketio XIO instance is created and handed to the user. So I really believe the name is good as is.